src/hooks/siteDashboardHooks/useGetSiteUrl.ts

```diff
@@ -0,0 +1,30 @@
+import { useQuery, UseQueryResult } from "react-query"
+
+import { SITE_URL_KEY } from "constants/queryKeys"
+
+import * as SiteDashboardService from "services/SiteDashboardService"
+
+const getSiteUrl = async (siteName: string): Promise<string> => {
+  return SiteDashboardService.getUserProvidedSiteUrl(siteName)
+    .then((url) => {
+      if (url) return `https://${url}`
+
+      return SiteDashboardService.getSiteUrl(siteName)
+    })
```

Comment:
Just curious: why do we prefer the user set url provided by settings here?

Reply:
Hmm it's because we should treat the user's provided URL as the truth (www.agency.gov.sg) and only fallback to the site URL stored inside the DB (master.gibberish.amplifyapp.com) if we don't have that information available.